File: archive-site-backend/src/ArchiveSiteBackend.Data/Document.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ArchiveSite.Data {
    public class Document : EntityBase<Document> {
        [Required, StringLength(1024)]
        public String FileName { get; set; }

        [Required, StringLength(1024)]
        public String DocumentImageUrl { get; set; }

        [Required]
        [ForeignKey(nameof(Project))]
        public Int64 ProjectId { get; set; }

        [Required]
        public DocumentStatus Status { get; set; } = DocumentStatus.PendingTranscription;

        [Required]
        public Int32 TranscriptionCount { get; set; }

        [ForeignKey(nameof(Transcription))]
        public Int64? ApprovedTranscriptionId { get; set; }

        public virtual Project Project { get; set; }
    }
}

File: archive-site-frontend/src/app/models/document.ts
import { Model } from './model'
import { Project } from 'src/app/models/project';

export class Document extends Model {
    constructor(
        id: number = 0,
        public ProjectId: number = undefined,
        public FileName: string = undefined,
        public DocumentImageUrl: string = undefined,
        public Status: string = "PendingTranscription",
        public Project?: Project) {
        super(id);
      }
}
